### PR TITLE
ENH: Add use_nullable_dtypes to to_numeric

### LINKED ISSUE #37294: ENH: Cast string dtype to nullable numeric types in to_numeric
https://github.com/pandas-dev/pandas/pull/37268 fixed a bug so that nullable string dtype can now be cast to numeric from "number-like" strings and NA, however the output dtype is currently float64. Better would be for this to cast to either nullable Int64 or Float64 depending on the values. Below is master:

```python
[ins] In [2]: import pandas as pd
         ...:
         ...: s = pd.Series(["1", "2", None], dtype="string")
         ...: print(pd.to_numeric(s))
         ...:
         ...: s = pd.Series(["1", "2.5", None], dtype="string")
         ...: print(pd.to_numeric(s))
         ...:
0    1.0
1    2.0
2    NaN
dtype: float64
0    1.0
1    2.5
2    NaN
dtype: float64
```

The first case would ideally cast to Int64, and the second to Float64.

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #37294 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I think to_numeric should be able to return nullable dtypes as well, similar to our I/O functions

### CODE REVIEW

**Review by mroeschke — COMMENTED** [review]
1. Could you add tests that exercises the `errors` keyword argument?

**Comment by phofl** [comment]
Thx, this found a couple of bugs

File: pandas/core/tools/numeric.py [hunk 1]
```diff
@@ -228,17 +246,28 @@ def to_numeric(
                     if values.dtype == dtype:
                         break
 
-    # GH33013: for IntegerArray & FloatingArray need to reconstruct masked array
-    if mask is not None:
+    # GH33013: for IntegerArray, BooleanArray & FloatingArray need to reconstruct
+    # masked array
+    if (mask is not None or new_mask is not None) and not is_object_dtype(values.dtype):
+        if mask is None:
+            mask = new_mask
```
Comment (jorisvandenbossche) [inline pandas/core/tools/numeric.py:253]:
```suggestion
        if mask is None:
            mask = new_mask
        else:
            mask = mask.copy()
```

and then remove the `.copy()` below when constructing the array (right now the mask is always copied, but assuming this newly generated mask doesn't need that)

**Review by jorisvandenbossche — COMMENTED** [review]
Can you add a test where `use_nullable_dtype=True` is specified but the input already is a nullable array?

File: pandas/core/tools/numeric.py (hunk 1 above)
Comment (phofl) [reply pandas/core/tools/numeric.py:253]: DOne

**Comment by phofl** [comment]
Added a test and found a problem for BooleanArray

**Review by mroeschke — APPROVED** [review]

**Comment by phofl** [comment]
Merging for now, happy to address additional comments in a follow up

**Pull request merged**